idaes/apps/uncertainty_propagation/uncertainties.py

```diff
@@ -0,0 +1,248 @@
+    
+    Args:
+        model_uncertain(function or Pyomo ConcreteModel) : Function is a a python Function that generates an instance of the Pyomo model
+        theta(dict)                    : Estimated parameters 
+        cov(numpy.ndarray)             : Covariance matrix of parameters 
+        theta_names(list of strings)   : List of estimated Var names
+        var_dic(dictionary)            : If any original variable contains apostrophes, need an auxiliary dictionary with keys theta_namess without apostrophes, values with apostrophes. e.g) var_dic: {'fs.properties.tau[benzene,toluene]': "fs.properties.tau['benzene','toluene']", 'fs.properties.tau[toluene,benzene]': "fs.properties.tau['toluene','benzene']"} 
+        tee(bool, optional)            : Indicates that ef solver output should be teed
+        solver_options(dict, optional) : Provides options to the solver (also the name of an attribute)
+
+     Returns:
+         tuple   : results object containing the all information including
+            - results.gradient_f(numpy.ndarray)    : Gradient vector of the objective function with respect to the (decision variables, parameters) at the optimal solution
+            - results.gradient_c(numpy.ndarray)    : Gradient vector of the constraints with respect to the (decision variables, parameters) at the optimal solution. Each row contains [column number, row number, and value], colum order follows variable order in col. If no constraint exists, return []
+            - results.dsdp(numpy.ndarray)          : Gradient vector of the (decision variables, parameters) with respect to paramerters (=theta_name). number of rows = len(theta_name), number of columns= len(col)
+            - results.propagation_c(numpy.ndarray) : Error propagation in the constraints (only constraints have theta_name)  e.g) [[constraint iumber1, error1],[constraint number2, error2]]             
+            - results.propagation_f(numpy.float64) : Error propagation in the objective function, df/dp*cov_p*df/dp + (df/dx*dx/dp)*cov_p*(df/dx*dx/dp)
+            - results.col(list)                    : List of variable names  
+    Raises:
+        Exception:  When model_uncertain is neither 'ConcreteModel' nor 'function'.
+    """
+
+    # define Pyomo model
+    if type(model_uncertain).__name__ == 'ConcreteModel':
+        model = model_uncertain
+    elif type(model_uncertain).__name__ == 'function':
+        model = model_uncertain()
+    else:
+        raise TypeError('model_uncertain must be either python function or Pyomo ConcreteModel.')
+    # Remove all "'" in theta_names     
+    theta_names, var_dic,variable_clean = clean_variable_name(theta_names)
+    for v in theta_names:
+        eval('model.'+var_dic[v]).setlb(theta[v])
```

Comment:
@dangunter Is there a better option besides `eval`?

Reply:
@Robbybp @dangunter @ksbeattie @jsiirola Can you provide a recommendation for what to use instead of `eval` here? Should it be ` getattr`?

Reply:
`model.getattr(var_dic[v]).setlb(theta[v])`

Reply:
replace `eval` to `model.find_component` based on @Robbybp's suggestion.  FYI:`getattr` not work.